pandas/core/frame.py

```diff
@@ -6259,12 +6259,13 @@ def explode(self, column: Union[str, Tuple]) -> "DataFrame":
         if not self.columns.is_unique:
             raise ValueError("columns must be unique")
 
-        result = self[column].explode()
-        return (
-            self.drop([column], axis=1)
-            .join(result)
-            .reindex(columns=self.columns, copy=False)
-        )
+        df = self.reset_index(drop=True)
+        result = df[column].explode()
+        result = df.drop([column], axis=1).join(result)
```

Comment:
I think the naming here is a bit confusing. `result` is first a `Series`, and then it's this `DataFrame`. I don't think the previous `result` is being named correctly.

Also thinking if this can be implemented with method chaining, I think that usually improves readability. I mean something like:
```python
df =self.reset_index(drop=True)
return (
    df.drop([column], axis=1)
      .join(df[column].explode())
      .set_index(self.index.take(df.index))
      .reindex(columns=self.columns, copy=False))
```

In any case, this is looking much better at every iteration.

Reply:
Yes, that's better - I still need to break it up slightly though, don't I? Else in `self.index.take(df.index)`, `df` isn't the right length.

Anyway, naming cleared up